### PR TITLE
ref(slack): condense block kit improvements changes under block kit flag

### PR DESCRIPTION
**Pull request by cathteng**
Removes usage of the `organizations:slack-block-kit-improvements` flag and merges the changes under that flag under the `organizations:slack-block-kit` flag.

### CODE REVIEW

File: tests/sentry/integrations/slack/test_message_builder.py [hunk 1]
```diff
@@ -888,64 +877,6 @@ def test_build_group_generic_issue_block(self):
...
-        for section in blocks["blocks"]:
-            if section["type"] == "text":
-                assert occurrence.issue_title in section["text"]["text"]
-
-        assert occurrence.evidence_display[0].value in blocks["blocks"][1]["text"]["text"]
-        assert blocks["text"] == f"[{self.project.slug}] {occurrence.issue_title}"
-
-    @with_feature("organizations:slack-block-kit")
-    @with_feature("organizations:slack-block-kit-improvements")
-    def test_build_group_generic_issue_block_no_escaping(self):
```
Comment (ceorourke) [inline tests/sentry/integrations/slack/test_message_builder.py:904]: why remove this test?

Comment (cathteng) [reply tests/sentry/integrations/slack/test_message_builder.py:904]: we don't escape anymore for block kit

Comment (cathteng) [reply tests/sentry/integrations/slack/test_message_builder.py:904]: oh wait

File: tests/sentry/integrations/slack/test_unfurl.py [hunk 2]
```diff
@@ -262,26 +262,6 @@ def test_escape_issue(self):
         unfurls = link_handlers[LinkType.ISSUES].fn(self.request, self.integration, links)
         assert unfurls[links[0].url]["text"] == "&amp;lt;https://example.com/|*Click Here*&amp;gt;"
 
-    @with_feature("organizations:slack-block-kit")
```
Comment (ceorourke) [inline tests/sentry/integrations/slack/test_unfurl.py:265]: also this one, do we not need escaping tests anymore?

File: tests/sentry/integrations/slack/test_message_builder.py (hunk 1 above)
Comment (cathteng) [reply tests/sentry/integrations/slack/test_message_builder.py:904]: this one is an oopsie

**Review by ceorourke — APPROVED** [review]

**Review by ykamo001 — APPROVED** [review]

**Review by isabellaenriquez — APPROVED** [review]

**Pull request merged**